### PR TITLE
Update to 3

### PR DESCRIPTION
**Pull request by allevo**
This PR changes the code in order to use mongodb@3

- the code is changed
- the tests are updated
- the README is updated

### CODE REVIEW

**Comment by allevo** [comment]
The latest commits introduce a new key.

The following registration
```
fastify.register(fastifyMongodb, { url: 'mongodb://localhost', database: 'mydb', name: 'my_conn_name' })
```
decorates the fastify instance with
```
{
  client: <Mongodb Client instance>,
  ObjectId: <Mongodb ObjectId class>,
  dbs: {
    mydb: <Mongodb Database instance>
  }
  my_conn_name: {
    client: <Mongodb Client instance>,
    ObjectId: <Mongodb ObjectId class>,
    dbs: {
      mydb: <Mongodb Database instance>
    }
    my_conn_name: [Circular]
  }
}
```

I don't know if this is a good thing. Thoughts?

File: index.js [hunk 1]
```diff
@@ -8,18 +8,23 @@ const ObjectId = MongoDb.ObjectId
...
+      client: client,
+      ObjectId: ObjectId,
+      dbs: {}
     }
     if (options.name) {
       mongo[options.name] = mongo
-      delete options.name
+    }
+    if (databaseName) {
+      mongo.dbs[databaseName] = client.db(databaseName)
```
Comment (mcollina) [inline index.js]: I think this is redundant, we already have `mongo[name] = mongo`

Comment (allevo) [reply index.js]: I don't understand what you mean.

The `name` parameter identifies a mongodb cluster, `database` parameter identifies a database on that cluster. Potentially you can have multiple databases assigned inside `mongo.dbs` (currently not implemented)

Comment (mcollina) [reply index.js]: The basic case is that you connect a single database. I see the value in providing `mongo.db` as it was before. Can you cover that case?

File: index.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ function fastifyMongodb (fastify, options, next) {
       mongo[options.name] = mongo
     }
     if (databaseName) {
-      mongo.dbs[databaseName] = client.db(databaseName)
+      mongo.dbs[databaseName] = mongo.db = client.db(databaseName)
```
[the remark this answers is not in the document]
Comment (allevo) [reply index.js]: Good. I'll remove the dbs key. I don't like complicate the code before there's no the need.
Anyway we could determine to which database the user would connect parsing the connection string. The path is the database on which you want to work

[the remark this answers is not in the document]
Comment (mcollina) [reply index.js]: I would prefer that too!

**Comment by allevo** [comment]
For interface see https://github.com/allevo/fastify-mongodb/blob/4cc104d433ae62944a47e3f175192460ef5316b1/test.js#L310-L350

**Comment by sebdeckers** [comment]
LGTM, works great. One thing I don't like about this plugin is how it mutates the `options` object. Is that necessary?

**Comment by mcollina** [comment]
@sebdeckers I'm going to land this. Would you like to send a PR that avoid mutation?

**Pull request merged**